enarx-keep/src/backend/kvm/vm/cpu.rs

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::syscall::SallyPort;
+use super::x86_64::*;
+
+use crate::backend::{Command, Thread};
+
+use enarx_keep_sev_shim::SYSCALL_TRIGGER_PORT;
+use kvm_ioctls::{VcpuExit, VcpuFd};
+use x86_64::registers::control::{Cr0Flags, Cr4Flags};
+use x86_64::registers::model_specific::EferFlags;
+use x86_64::PhysAddr;
+
+use std::convert::AsMut;
+use std::io::{Error, ErrorKind, Result};
+
+pub struct Cpu {
+    fd: VcpuFd,
+    sallyport: SallyPort,
```

Comment:
I'm surprised that `Cpu` doesn't keep an `Arc<Mutex<VirtualMachine>>`.

Reply:
Why not store the `sallyport::Block` directly here?

Reply:
> I'm surprised that `Cpu` doesn't keep an `Arc<Mutex<VirtualMachine>>`.

I think it will need to grow to do so when we start hotplugging vCPUs.

> Why not store the `sallyport::Block` directly here?

I think this leaves us some flexibility in the event the shim decides to relocate the shared pages alongside itself but this is not a use case that any of us have expressed yet. Aside from that, there's no technical benefit from the way I did it.